src/frontend/next/src/pages/search.tsx

```diff
@@ -0,0 +1,67 @@
+import { FormEvent, useState } from 'react';
+import { makeStyles } from '@material-ui/core/styles';
+import { useRouter } from 'next/router';
+
+import SearchResults from '../components/SearchResults';
+import SearchBar from '../components/SearchBar';
+import BackToTopButton from '../components/BackToTopButton';
+
+enum FilterParamType {
```

Comment:
Do it like https://github.com/Seneca-CDOT/telescope/blob/master/src/frontend/next/src/components/SearchResults.tsx#L45, and in a follow-up, we can synchronize all of these to use a common definition of the filter params.

Reply:
Ah so something like  
```
type FilterProp = {
  filter: 'post' | 'author';
};
```

and to type the hooks I use
```
const [filter, setFilter] = useState<FilterProp['filter']>('post');
```